hack/generate/gen-test-framework.sh

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+
+#############################
+# Make sure that test/test-framework is updated according to the current source code
+#############################
+
+set -ex
+
+source hack/lib/test_lib.sh
+
+# Define vars
+DOCKERFILE="build/Dockerfile"
+
+# Go inside of the mock data test project
+cd test/test-framework
+
+# Create the required files in the mock test/test-framework just to allow run from it the oprator-sdk commands
+echo > $DOCKERFILE
+go mod init test-framework
+
+# Remove files that are required and created just to exec the commands.
+trap_add 'rm -rf $DOCKERFILE go.mod go.sum' EXIT
+
+# Run gen commands
+operator-sdk generate k8s
+operator-sdk generate openapi
+
+# The following file is gen by openapi but it has not been commited in order to allow we clone and call the test locally in any path. 
+trap_add 'rm -rf pkg/apis/cache/v1alpha1/zz_generated.openapi.go' EXIT
```

Comment:
A few nits:
1. Add a `TODO(camilamacedo86)` in the comment to remove this when we remove the `zz_generated.openapi.go` generation from `operator-sdk generate openapi`
2. Change to:
    ```
    trap_add 'rm pkg/apis/cache/v1alpha1/zz_generated.openapi.go' EXIT
    ```
    That way, the `rm` command will fail if that file doesn't exit. It'll be a reminder to go fix this when we stop generating it.